libs/auth/react/src/lib/auth/AuthProvider.tsx

```diff
@@ -0,0 +1,260 @@
+        return
+      }
+
+      // If all else fails, redirect to the login page.
+      await signIn()
+    } else {
+      // When not performing autologin, silently check if there's an IDP session.
+      await signInSilent()
+    }
+  }, [userManager, dispatch, signIn, signInSilent, autoLogin])
+
+  useEffect(() => {
+    // Only add events when we have userInfo, to avoid race conditions with
+    // oidc hooks.
+    if (state.userInfo === null) {
+      return
+    }
+
+    // This is raised when a new user state has been loaded with a silent login.
+    const userLoaded = (user: User) => {
+      dispatch({
+        type: ActionType.USER_LOADED,
+        payload: user,
+      })
+    }
+
+    // This is raised when the user is signed out of the IDP.
+    const userSignedOut = async () => {
+      dispatch({
+        type: ActionType.LOGGED_OUT,
+      })
+      await userManager.removeUser()
+
+      if (autoLogin) {
+        signIn()
+      }
+    }
+
+    userManager.events.addUserLoaded(userLoaded)
+    userManager.events.addUserSignedOut(userSignedOut)
+    return () => {
+      userManager.events.removeUserLoaded(userLoaded)
+      userManager.events.removeUserSignedOut(userSignedOut)
+    }
+  }, [dispatch, userManager, signIn, autoLogin, state.userInfo === null])
+
+  const isCurrentRoute = (path?: string) =>
+    isDefined(path) && urlRef.current?.includes(basePath + path)
+
+  const init = async () => {
+    if (isCurrentRoute(authSettings?.redirectPath)) {
+      try {
+        const user = await userManager.signinRedirectCallback(
+          window.location.href,
+        )
+
+        dispatch({ type: ActionType.SIGNIN_SUCCESS, payload: user })
+
+        const url = typeof user.state === 'string' ? user.state : '/'
+        window.location.replace(url)
+      } catch (error) {
+        if (error.error === 'login_required') {
+          // If trying to switch delegations and the IDS session is expired, we'll
+          // see this error. So we'll try a proper signin.
+          return userManager.signinRedirect({ state: error.state })
+        }
+        console.error('Error in oidc callback', error)
+        setHasError(true)
+      }
+    } else if (isCurrentRoute(authSettings?.redirectPathSilent)) {
+      const userManager = getUserManager()
+      userManager.signinSilentCallback().catch((error) => {
+        // TODO: Handle error
+        console.log(error)
+      })
+    } else {
+      checkLogin()
+    }
+  }
+
+  useEffect(() => {
+    init()
+  }, [])
+
+  const context = useMemo(
+    () => ({
+      ...state,
+      signIn,
+      signInSilent,
+      switchUser,
+      signOut,
+    }),
+    [state, signIn, signInSilent, switchUser, signOut],
+  )
+
+  const isLoading =
+    !state.userInfo ||
+    // We need to display loading screen if current route is the redirect path.
+    // This is because the redirect path in not part of our React Router routes.
+    isCurrentRoute(authSettings?.redirectPath)
```

Comment:
How about redirectPathSilent? Is userInfo ever set on these routes?

The redirectPathSilent stuff happens in an iframe. Should make sure we don't render the whole app in the iframe.